zkvm/src/vm.rs

```diff
@@ -115,6 +119,8 @@ pub trait Delegate<CS: r1cs::ConstraintSystem> {
     /// Returns Ok(None) if there is no more instructions to execute.
     fn next_instruction(&mut self, run: &mut Self::RunType)
         -> Result<Option<Instruction>, VMError>;
+
+    fn next_run(&mut self, prog: &[u8]) -> Self::RunType;
```

Comment:
`next_run` is a bit misleading - it implies that the creating the run has sideeffects, while it should not - VM manages the runstack, not the delegate. How about:

```suggestion
    fn new_run(&self, prog: &[u8]) -> Self::RunType;
```

(note the self is immutably borrowed)